framework/include/actions/Action.h

```diff
@@ -44,8 +46,48 @@ class Action : public ConsoleStreamInterface
 
   virtual void act() = 0;
 
+  /**
+   * The name of the action
+   *
+   * When using MOOSE input files, this is the last portion
+   * of the full input syntax:
+   *    [System]
+   *      [./name]
+   *
+   *    [System]
+   *      [./Subsystem]
+   *        [./name]
+   */
   const std::string & name() const { return _name; }
 
+  ///@{
+  /**
+   * The full input file and base syntax for the action
+   *
+   * _full_syntax = System/name
+   * _base_syntax = System
+   * [System]
+   *   [./name]
+   *
+   * _full_syntax = System/Subsystem/name
+   * _base_syntax = System/Subsystem
+   * [System]
+   *   [./Subsystem]
+   *     [./name]
+   */
+  const std::string & getInputSyntax() const { return _full_syntax; }
```

Comment:
This stuff makes me throw up in my mouth a little bit (that doesn't mean I'm _against_ it... just that it's unpalatable).

@permcody has worked hard to keep syntax _out_ of Actions.  We may need to think about this some more (once you go down this path you can't ever go backward).